flink-runtime/src/main/java/org/apache/flink/runtime/operators/resettable/SpillingResettableIterator.java

```diff
@@ -189,4 +189,12 @@ public void remove() {
 			return memory;
 		}
 	}
+
+	private void increaseElementCount() {
```

Comment:
This method is a bit overengineered: even if we are doing nothing than just incrementing the long value, it takes 60 years on a 4 GHz CPU to reach the overflow in java. 
Nevertheless, I expect the branch prediction to also figure that out and not cause any performance degradation.

Reply:
I agree with you, it was realy a bit overengineered, so I removed the logic of checking long overflow.